### PR TITLE
Updated README with gifs

### PR DESCRIPTION
**Pull request by m13467**
I just edited the README from the suggestions that Dr. Bean gave. I added a few gifs, which we may have to change, but then we'll know how it looks better. I uploaded those gifs to the repo. I also added a doc that explains the local setup.

### CODE REVIEW

**Review by EianWright — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -1,50 +1,39 @@
...
-
-### Code Review
-
-This extension has the option of reviewing a whole file or reviewing selected code. When it reviews your code, it will look for bugs in the code and look for general naming problems. In addition, it will suggest getting rid of or adding spacing to make the code easier to read. When you ask it to review your code, it will open another tab showing a dif view of your code and the code that it suggests. You can then choose which changes you will add to your code. 
-
-### Code Completion
+In addition, byoLAD enables you to use your own LLM API, so you can use your favorite AI and control how much you spend. We don't add any extra costs, and there are no extra parties looking at your data. You only have to manage your account with your LLM provider. 
+#### LLM Providers Currently Supported: 
+- OpenAI
+- Google
```
Comment (byubean) [inline README.md:8]: Let's add links here to these providers

File: README.md [hunk 2]
```diff
@@ -1,50 +1,39 @@
...
-The only requirement needed if not running locally would be to set your API key in the setting, which is explained in this document. You will have to get your own API key for the AI service you would like to use like OpenAI or ChatGPT. 
+## Features
 
-<a target="_blank" href="https://code.visualstudio.com/docs/getstarted/settings">How to set your API key</a>
+- Option to review a whole file or just a piece of selected code
+- Formatting suggestions like getting rid of extra lines to make it easier to read
+- Identifying bugs in code
+- Adds Documentation
+- Checks Spelling
+- General improvement suggestions
```
Comment (byubean) [inline README.md:19]: Standardize the verb tenses to 3rd person present (e.g. "Suggests formatting improvements", "Identifies bugs", etc)

File: README.md [hunk 3]
```diff
@@ -1,50 +1,39 @@
...
-3. Copy the .vscode.sample folder and rename it to .vscode.
+![](SetAPIKey.gif)
 
-#### To verify that it's working:
-1. With VS code open, press F5.
-2. Another "Extension Development Host" VS code window should open.
-3. With a file open, right click.
-4. Click "Review Code File".
-5. An error message may pop up saying "APIKey not set". If not, then it should have opened a new tab showing your code and the suggested changes.
+> Note: If you want to run the extension locally, go to the [contribute.md](contribute.md)
```
Comment (byubean) [inline README.md:33]: "go to ~the~ [contribute]..."

File: README.md [hunk 4]
```diff
@@ -1,50 +1,39 @@
...
 
-> Note: The instructions for setting your APIKey in setting is <a target="_blank" href="https://code.visualstudio.com/docs/getstarted/settings">here</a>
+## Who is the BeanLab?
 
-#### Strongly suggested options
-- Install the vscode-eslint and prettier-vscode extensions.
-- Set Prettier as your default formatter in VS code settings.
-- Turn on format on save in VS code settings.
+We are a lab at Brigham Young University, and our goal is to provide useful open-source programs for everyone to use. We are made up of a small group of students and a professor. We enjoy pushing ourselves to make new and improved programs, and we are happy to share them with everyone.
```
Comment (byubean) [inline README.md:37]: Drop the sentence "We are made up...."

Comment (byubean) [reply README.md:37]: I think we'll probably replace this whole section with a logo that links to a general Bean Lab page (that I haven't written yet...)

File: ReviewCode.gif
Comment (byubean) [inline ReviewCode.gif:1]: what model are you using here? GPT-4? If so, try GPT-3.5-turbo (it goes 10x faster and may be better for a visual demo)

File: SetAPIKey.gif
Comment (byubean) [inline SetAPIKey.gif:1]: We'll need to change the extension name to "byoLAD" and redo this shot. And once we change how you select/specify the model we'll change it again. So change it yet (this comment is just a reminder)

File: contribute.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
+# Contribute
+
+Please feel free to contribute to our project. Here are the instructions for anyone who wants to run this extension locally. 
+
+## Local Set Up
+
+If you desire to run this extension locally, here are the steps to have it run properly.
```
Comment (byubean) [inline contribute.md:7]: Drop this line

File: contribute.md [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+Please feel free to contribute to our project. Here are the instructions for anyone who wants to run this extension locally. 
+
+## Local Set Up
+
+If you desire to run this extension locally, here are the steps to have it run properly. 
+
+#### Specific instructions for setup:
+1. Clone the repo.
+2. Run npm install.
+3. Copy the .vscode.sample folder and rename it to .vscode.
```
Comment (byubean) [inline contribute.md:12]: Is this still needed?

File: contribute.md [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+
+#### To verify that it's working:
+1. With VS code open, press F5.
+2. Another "Extension Development Host" VS code window should open.
```
Comment (byubean) [inline contribute.md:16]: ~Another~ An "Extension..."

File: contribute.md [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+3. With a file open, right click.
```
Comment (byubean) [inline contribute.md:17]: Right click an open file in the editor

File: contribute.md [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+4. Click "Review Code File".
+5. An error message may pop up saying "APIKey not set". If not, then it should have opened a new tab showing your code and the suggested changes.
```
Comment (byubean) [inline contribute.md:19]: Add a new step 3 that sets the API key in the configuration so this error doesn't happen. Then drop this item.

File: README.md (hunk 4 above)
Comment (m13467) [reply README.md:37]: Do you have a Bean Lab logo? If so, we could just put it in for now, and then add the link to the image later when you have written the page for the Bean Lab.

File: ReviewCode.gif
Comment (m13467) [reply ReviewCode.gif:1]: I am using GPT-3.5-turbo I think.

File: contribute.md (hunk 6 above)
Comment (m13467) [reply contribute.md:12]: I don't think so. I'll remove it.

**Pull request merged**